tests/test_business.py

```diff
@@ -410,3 +416,83 @@ def test_equality(self):
 
         assert boh1 != boh3
         assert hash(boh1) != hash(boh3)
+
+    def test_get_opening_hours_for_day_without_timezone(
+        self, business_opening_hours: BusinessOpeningHours
+    ):
+        target_date = date(2023, 7, 17)  # Monday
+        target_date_plus_1 = date(2023, 7, 18)  # Tuesday
+        expected_tz = zoneinfo.ZoneInfo(business_opening_hours.time_zone_name)
+        expected_opening_time = datetime(2023, 7, 17, 7, 0, tzinfo=expected_tz)
+        expected_closing_time = datetime(2023, 7, 17, 22, 0, tzinfo=expected_tz)
+
+        opening_hours_for_day = business_opening_hours.get_opening_hours_for_day(target_date)
+
+        assert opening_hours_for_day[0][0] == expected_opening_time
+        assert opening_hours_for_day[0][1] == expected_closing_time
+
+        assert opening_hours_for_day[0][0].date() != target_date_plus_1
+        assert opening_hours_for_day[0][1].date() != target_date_plus_1
+        assert opening_hours_for_day[1][0].date() != target_date_plus_1
+        assert opening_hours_for_day[1][1].date() != target_date_plus_1
+
+    def test_get_opening_hours_for_day_with_timezone(
+        self, business_opening_hours: BusinessOpeningHours
+    ):
+        target_date = date(2023, 7, 17)  # Monday
+        target_date_plus_1 = date(2023, 7, 18)  # Tuesday
+        test_tz = zoneinfo.ZoneInfo("UTC")
+        expected_opening_time = datetime(2023, 7, 17, 7, 0, tzinfo=test_tz)
+        expected_closing_time = datetime(2023, 7, 17, 22, 0, tzinfo=test_tz)
+
+        opening_hours_for_day = business_opening_hours.get_opening_hours_for_day(
+            target_date, tzinfo=test_tz
+        )
+
+        assert opening_hours_for_day[0][0] == expected_opening_time
+        assert opening_hours_for_day[0][1] == expected_closing_time
+
+        assert opening_hours_for_day[0][0].date() != target_date_plus_1
+        assert opening_hours_for_day[0][1].date() != target_date_plus_1
+        assert opening_hours_for_day[1][0].date() != target_date_plus_1
+        assert opening_hours_for_day[1][1].date() != target_date_plus_1
```

Comment:
`opening_hours_for_day` must have only one entry - you're requesting tha opening slots for monday and there is only one `BusinessOpeningHoursInterval` defined for monday